src/document.rs

```diff
@@ -693,87 +416,42 @@ fn get_code_block_annotation(node: &Node) -> HowserResult<MatchType> {
 
 /// Strips the annotation from a code block node if one exists.
 fn remove_code_block_annotation(node: &Node) -> HowserResult<()> {
-    let info = node.capabilities
-        .get
-        .as_ref()
-        .ok_or(HowserError::CapabilityError)?
-        .get_fence_info()?;
-    let (match_type, content) = extract_match_type(&info)?;
-    if match_type != MatchType::None {
-        node.capabilities
-            .set
-            .as_ref()
-            .ok_or(HowserError::CapabilityError)?
-            .set_fence_info(&content)?;
+    if let Node::CodeBlock(code_block) = node {
+        let (match_type, content) = extract_match_type(&code_block.get_fence_info()?)?;
+        if match_type != MatchType::None {
+            code_block.set_fence_info(&content)?;
+        }
     }
 
     Ok(())
 }
 
 /// Returns the block-level annotation of a heading node if one exists.
-fn get_heading_annotation(node: &Node) -> HowserResult<MatchType> {
-    if let Some(text) = node.capabilities
-        .traverse
-        .as_ref()
-        .ok_or(HowserError::CapabilityError)?
-        .first_child()?
-    {
-        if text.capabilities
-            .traverse
-            .as_ref()
-            .ok_or(HowserError::CapabilityError)?
-            .next_sibling()?
-            .is_none()
-        {
-            let getter = text.capabilities
-                .get
-                .as_ref()
-                .ok_or(HowserError::CapabilityError)?;
-            if getter.get_type()? == NodeType::CMarkNodeText {
-                let (match_type, content) = extract_match_type(&getter.get_content()?)?;
+fn get_heading_annotation(heading_node: &Node) -> HowserResult<MatchType> {
+    if let Some(text_node) = heading_node.first_child()? {
+        if let Node::Text(ref text) = text_node {
+            if text_node.next_sibling()?.is_none() {
```

Comment:
This nesting is getting kind of deep; I wonder if you can flatten it by inverting the conditions and returning early? Might not work with the `if let` idiom.